### PR TITLE
fix: Custom JSON encoder for List and Map doesn't work correctly in `full` mode

### PR DESCRIPTION
**Pull request by mbfakourii**
### New Pull Request Checklist
- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/Parse-SDK-Flutter/security/policy).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/Parse-SDK-Flutter/issues?q=is%3Aissue).

### Issue Description
Solve the parseEncode problem in list and map
and test added
Related issue: https://github.com/parse-community/Parse-SDK-Flutter/issues/788

### Approach
The ```full: full``` was added for when we use the list and map in parseEncode and toJson

### TODOs before merging

- [ ] Add tests
- [ ] Add changes to documentation (guides, repository pages, in-code descriptions)
- [ ] A changelog entry

### CODE REVIEW

**Comment by mtrezza** [comment]
Could you please rephrase the PR title? It will become the changelog entry and a developer should be able to understand what this fixes without reading the PR.

File: packages/dart/lib/src/utils/parse_encoder.dart [hunk 1]
```diff
@@ -20,13 +20,13 @@ dynamic parseEncode(dynamic value, {bool full = false}) {
 
   if (value is List) {
     return value.map<dynamic>((dynamic value) {
-      return parseEncode(value);
+      return parseEncode(value, full: full);
```
Comment (mtrezza) [inline packages/dart/lib/src/utils/parse_encoder.dart:23]:
```suggestion
      return parseEncode(value, full);
```

File: packages/dart/lib/src/utils/parse_encoder.dart [hunk 2]
```diff
@@ -20,13 +20,13 @@ dynamic parseEncode(dynamic value, {bool full = false}) {
...
     }).toList();
   }
 
   if (value is Map) {
     value.forEach((dynamic k, dynamic v) {
-      value[k] = parseEncode(v);
+      value[k] = parseEncode(v, full: full);
```
Comment (mtrezza) [inline packages/dart/lib/src/utils/parse_encoder.dart:29]:
```suggestion
      value[k] = parseEncode(v, full);
```

**Review by mtrezza — CHANGES_REQUESTED** [review]
Could you add a test to demonstrate what is not working correctly?

**Comment by mbfakourii** [comment]
> Could you add a test to demonstrate what is not working correctly?

Test added

File: packages/dart/lib/src/utils/parse_encoder.dart [hunk 3]
```diff
@@ -20,7 +20,7 @@ dynamic parseEncode(dynamic value, {bool full = false}) {
 
   if (value is List) {
     return value.map<dynamic>((dynamic value) {
-      return parseEncode(value, full: full);
+      return parseEncode(value, full);
```
[the remark this answers is not in the document]
Comment (mtrezza) [reply packages/dart/lib/src/utils/parse_encoder.dart:23]: Apologies, I was thinking in the wrong programming language

File: packages/dart/lib/src/utils/parse_encoder.dart [hunk 4]
```diff
@@ -20,13 +20,13 @@ dynamic parseEncode(dynamic value, {bool full = false}) {
 
   if (value is List) {
     return value.map<dynamic>((dynamic value) {
-      return parseEncode(value);
+      return parseEncode(value, full);
```
Comment (mtrezza) [inline packages/dart/lib/src/utils/parse_encoder.dart:23]:
```suggestion
      return parseEncode(value, full: full);
```

File: packages/dart/lib/src/utils/parse_encoder.dart [hunk 5]
```diff
@@ -20,13 +20,13 @@ dynamic parseEncode(dynamic value, {bool full = false}) {
...
     return value.map<dynamic>((dynamic value) {
-      return parseEncode(value);
+      return parseEncode(value, full: full);
     }).toList();
   }
 
   if (value is Map) {
     value.forEach((dynamic k, dynamic v) {
-      value[k] = parseEncode(v);
+      value[k] = parseEncode(v, full);
```
Comment (mtrezza) [inline packages/dart/lib/src/utils/parse_encoder.dart:29]:
```suggestion
      value[k] = parseEncode(v, full: full);
```

**Review by mtrezza — COMMENTED** [review]
There's still a lint issue, not sure what it is, because the logs don't show any specific message. Do you have any idea?

**Comment by mbfakourii** [comment]
> There's still a lint issue, not sure what it is, because the logs don't show any specific message. Do you have any idea?

It is for Dart code format
I made a commit, it is probably fixed

**Comment by mtrezza** [comment]
Indeed, the tests pass, now we just need a review from @parse-community/flutter-sdk-review

**Comment by mtrezza** [comment]
There has been no additional review, but I'll go ahead and merge it anyway; if there is an issue, we'll find out.

**Review by mtrezza — APPROVED** [review]
Looks good!

**Pull request merged**